dh_commitments/src/afgho16/mod.rs

```diff
@@ -0,0 +1,93 @@
+use algebra::curves::PairingEngine;
+use rand::Rng;
+use std::marker::PhantomData;
+
+use crate::{random_generators, DoublyHomomorphicCommitment, Error};
+
+use inner_products::{ExtensionFieldElement, InnerProduct, PairingInnerProduct};
+
+#[derive(Clone)]
+pub struct AFGHOCommitment<P: PairingEngine> {
+    _pair: PhantomData<P>,
+}
+
+#[derive(Clone)]
+pub struct AFGHOCommitmentG1<P: PairingEngine>(AFGHOCommitment<P>);
+
+#[derive(Clone)]
+pub struct AFGHOCommitmentG2<P: PairingEngine>(AFGHOCommitment<P>);
```

Comment:
Do these need to be separate types, beyond, eg: type safety? Or can you just do something like this:
```suggestion
pub type AFGHOCommitmentG1<P> = AFGHOCommitment<P>;

pub struct AFGHOCommitmentG2<P> = AFGHOCommitment<P>;
```